pkg/keptn/resource_service.go

```diff
@@ -51,7 +52,47 @@ func (r V1ResourceHandler) GetResource(resource string, gitCommitID string) ([]b
 		queryParam = api.AppendQuery(url.Values{})
 	}
 
-	resourceContent, err := r.ResourceHandler.GetResource(*scope, queryParam)
+	return queryParam
+}
+
+// GetServiceResource returns the contents of a resource for a given gitCommitID
+func (r V1ResourceHandler) GetServiceResource(resource string, gitCommitID string) ([]byte, error) {
+	scope := api.NewResourceScope()
+	scope.Service(r.Event.Service)
+	scope.Project(r.Event.Project)
+	scope.Stage(r.Event.Stage)
+	scope.Resource(resource)
+
+	resourceContent, err := r.ResourceHandler.GetResource(*scope, buildResourceHandlerV1Options(gitCommitID))
+	if err != nil {
+		return nil, fmt.Errorf("unable to get resouce from keptn: %w", err)
+	}
+
+	return []byte(resourceContent.ResourceContent), nil
+}
+
+// GetProjectResource returns the resource that was defined on project level
+func (r V1ResourceHandler) GetProjectResource(resource string, gitCommitID string) ([]byte, error) {
```

Comment:
We could remove the `gitCommitID` parameter since it does not work as you mentioned. If not, we should probably document it in the function comment.

Reply:
`gitCommitID` works fine on Keptn side, so it should stay in here.
though when WE use it as part of a task/sequence, the git commit id always refers to service/stage level, not project level. therefore we ommit the parameter in our calling code, but not in the actual implementation.

Reply:
Ok, I thought that we are always calling it with `gitCommitID=""`  because of that. If that is not the case, we can leave it as is.